front/components/assistant/conversation/ActionValidationProvider.tsx

```diff
@@ -108,71 +94,59 @@ export function ActionValidationProvider({
 
   useNavigationLock(isDialogOpen);
 
-  const sendCurrentValidation = useCallback(
-    async (status: MCPValidationOutputType) => {
-      if (!currentValidation) {
-        return;
-      }
-
-      let approved = status;
-      if (status === "approved" && neverAskAgain) {
-        approved = "always_approved";
-      }
+  const sendCurrentValidation = async (status: MCPValidationOutputType) => {
```

Comment:
I don't know why it's wrapped with `useCallback`, seems unnecessary so removed it